tide_site_restriction.module

```diff
@@ -162,3 +158,81 @@ function tide_site_restriction_form_views_exposed_form_alter(&$form, FormStateIn
     }
   }
 }
+
+/**
+ * Implements hook_node_access_records().
+ */
+function tide_site_restriction_node_access_records(Node $node) {
+  $grants = [];
+  if (!$node->isPublished()) {
+    $sites_values = $node->get('field_node_site')->getValue();
+    if (empty($sites_values)) {
+      return $grants;
+    }
+    $sites = array_column($sites_values, 'target_id');
+    foreach ($sites as $site) {
+      $grants[] = [
+        'realm' => 'tide_site_restriction',
+        'gid' => $site,
+        'grant_view' => 1,
+        'grant_update' => 0,
+        'grant_delete' => 0,
+        'nid' => $node->id(),
+      ];
+    }
+  }
+  return $grants;
+}
+
+/**
+ * Implements hook_node_grants().
+ */
+function tide_site_restriction_node_grants(AccountInterface $account, $op) {
+  $user = User::load($account->id());
+  $user_sites = $user->get('field_user_site')->getValue();
```

Comment:
On line 224 you use `$user_sites = $helper->getUserSites($user);` and check if user can bypass restriction. Why is this line different and do you need to check bypass here?

Reply:
Hi @anthony-malkoun , 
1. The permission check on line 224 will not apply to `views`. the hook`hook_node_grants` is for the view to print the correct result.

2. line 224 does actual permission check based on operations.